hazelcast/src/main/java/com/hazelcast/config/DynamicClassloadingConfig.java

```diff
@@ -0,0 +1,235 @@
+         * but they won't be served to other members.
+         *
+         */
+        LOCAL_CLASSES_ONLY,
+
+        /**
+         * Server classes loaded from both local classpath and from other members.
+         *
+         */
+        LOCAL_AND_CACHED_CLASSES
+    }
+
+    private ClassCacheMode classCacheMode = ClassCacheMode.ETERNAL;
+    private ProviderMode providerMode = ProviderMode.LOCAL_AND_CACHED_CLASSES;
+    private String blacklistedPrefixes;
+    private String whitelistedPrefixes;
+    private String providerFilter;
+    private boolean enabled;
+
+    /**
+     * Enable or disable dynamic classloading. Default: {{@code false}}
+     *
+     * @param enabled
+     * @return
+     */
+    public DynamicClassloadingConfig setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    /**
+     * Returns true when dynamic classloading is enabled
+     *
+     * @return {{@code true}} when dynamic classloading is enabled
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Filter to constraint members to be used for classloading request when a class
+     * is not available locally.
+     *
+     * Filter format: <code>HAS_ATTRIBUTE:foo</code> this will send classloading requests
+     * only to members which has a member attribute <code>foo</code> set. Value is ignored,
+     * it can be any type. A present of the attribute is sufficient.
+     *
+     * This facility allows to have a control on classloading. You can e.g. start Hazelcast lite
+     * members dedicated for class-serving.
+     *
+     * Setting the filter to null will allow to load classes from all members.
+     *
+     * Default: {{@code null}}
+     *
+     *
+     * @see {@link com.hazelcast.core.Member#setStringAttribute(String, String)}
+     * @param providerFilter
+     * @return this instance of DynamicClassloadingConfig for easy method-chaining.
+     */
+    public DynamicClassloadingConfig setProviderFilter(String providerFilter) {
+        this.providerFilter = providerFilter;
+        return this;
+    }
+
+    /**
+     * Get current filter or {{@code null}} when no filter is defined.
+     *
+     * @see #setProviderFilter(String)
+     * @return current filter or {{@code null}} when no filter is defined.
+     */
+    public String getProviderFilter() {
+        return providerFilter;
+    }
+
+    /**
+     * Comma separated list of prefixes of classes which will never be loaded remotely.
+     * A prefix can be a package name or a classname.
+     *
+     * For example setting a blacklist prefix to <code>com.foo</code> will disable remote loading of all classes
+     * from the <code>com.foo</code> package, but also classes from all sub-packages won't be loaded.
+     * Eg. <code>com.foo.bar.MyClass</code> will be black-listed too.
+     *
+     * When you set the blacklist to <code>com.foo.Class</code> then the Class will obviously be black-listed,
```

Comment:
You can't. The `.class` prefix is an internal representation of classes on a filesystem. 
However here you are specifiying logical classname prefixes. 

I understand your point, perhaps in the future we could add "strict-match-mode" instead prefix-match.